### ISSUE TITLE
Random crash SIGABRT

### ISSUE THREAD

**Issue by shekohex** [zed-industries/zed#7509]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After running zig on a big rust project on my mac M1 PRO, it crashed with `SIGABRT` SIGNAL

### Environment

Zed: v0.120.6 (Zed)
OS: macOS 14.3.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Here is macOS crash report: https://gist.github.com/shekohex/24f5ed17c4deae49b6b9fa037fff1669

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[Zed.log](https://github.com/zed-industries/zed/files/14197065/Zed.log)

**Comment by ConradIrwin** [comment]
Thanks for reporting this! SIGABRT happens on panic, and it looks like this corresponds to the following panic we recorded internally.

```
thread 'main' panicked at 'assertion `left == right` failed: invalid insertion
  left: Lamport {0: 1}
 right: Lamport {0: 100}'
crates/text/src/text.rs:1883
<backtrace::capture::Backtrace>::create
<backtrace::capture::Backtrace>::new
Zed::init_panic_hook::{closure#0}
std::panicking::rust_panic_with_hook
std::panicking::begin_panic_handler::{{closure}}
std::sys_common::backtrace::__rust_end_short_backtrace
_rust_begin_unwind
core::panicking::panic_fmt
core::panicking::assert_failed_inner
core::panicking::assert_failed::<clock::Lamport, clock::Lamport>
<text::BufferSnapshot>::summary_for_anchor::<usize>
<editor::Editor>::rename::{closure#1}::{closure#0}
<async_task::raw::RawTask<<async_task::runnable::Builder<_>>::spawn_local::Checked<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output = core::result::Result<(), anyhow::Error>>>>>, core::result::Result<(), anyhow::Error>, <gpui::executor::ForegroundExecutor>::spawn::inner<core::result::Result<(), anyhow::Error>>::{closure#0}, ()>>::run
<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run
Zed::main
std::sys_common::backtrace::__rust_begin_short_backtrace::<fn(), ()>
std::rt::lang_start::<()>::{closure#0}
std::rt::lang_start_internal
std::rt::lang_start::<()>
_main
```

We've been trying to track this one down for a while, but no luck yet. Do you happen to remember what you were doing at the time of the crash?

**Comment by shekohex** [comment]
So I remember trying to rename a function (F2) and it freezed for a bit then crashed.

**Comment by JosephTLyons** [comment]
This has landed in [v0.123.2](https://zed.dev/releases/stable/0.123.2).
